File: beginner/__1000__/1021/1021.lua
local notas = {10000, 5000, 2000, 1000, 500, 200}
local moedas = {100, 50, 25, 10, 5, 1}

local reais, centavos = io.read():match("(%d+)%.(%d+)")
reais, centavos = tonumber(reais), tonumber(centavos)

reais = reais * 100 + centavos

print("NOTAS:")
for _, value in ipairs(notas) do
    local n = reais // value
    print(string.format("%d nota(s) de R$ %.2f", n, value / 100))
    reais = reais % value
end

print("MOEDAS:")
for _, value in ipairs(moedas) do
    local m = reais // value
    print(string.format("%d moeda(s) de R$ %.2f", m, value / 100))
    reais = reais % value
end

File: beginner/__1000__/1021/1021.ts
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin", 'utf-8');

let [reais, centavos] = input.split(".").map(Number);

// Converte tudo para centavos
reais = reais * 100 + centavos;

const notas: number[]  = [10000, 5000, 2000, 1000, 500, 200];
const moedas: number[] = [100,   50,   25,   10,   5,   1  ];

console.log("NOTAS:");
for (const nota of notas) {
    console.log(`${Math.floor(reais / nota)} nota(s) de R$ ${(nota / 100).toFixed(2)}`);
    reais %= nota;
}

console.log("MOEDAS:");
for (const moeda of moedas) {
    console.log(`${Math.floor(reais / moeda)} moeda(s) de R$ ${(moeda / 100).toFixed(2)}`);
    reais %= moeda;
}
